src/features/pastelExpertConsole/rpc-services/api.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import axios from 'axios'
+
+import store from '../../../redux/store'
+import { APIMethods, METHODS } from './utils'
+
+const getRPCConfig = () => {
+  const state = store.getState()
+  const { pastelConf } = state
+
+  return pastelConf || {}
+}
+
+const getMessage = (statusCode: number, isECONNREFUSED: boolean) => {
+  if (isECONNREFUSED) {
+    return 'Pastel could not find a daemon running, please check the logs!'
+  }
+
+  switch (statusCode) {
+    case 401:
+      return 'Not authorized to access RPC, please check your rpcuser and rpcpassword'
+    default:
+      return 'Something went wrong'
+  }
+}
+
+const apiMethods: any = {}
```

Comment:
Please think about a type that is better than "any". This could be:
```ts
type possibleMethods = typeof METHODS[number]
type dynamicAPI = {
  [x in possibleMethods]: <T>(params: string[]) => Promise<T>
}

const apiMethods = {} as dynamicAPI
```
And of course, you can select your own names :)

Now, we have a fully typed `apiMethods` object, that can be called like that:
```ts
const r = apiMethods('z_importwallet')<TGetBlockResult>(['hash']) // TGetBlockResult is the new type from broken down APIMethods type. 
```
The `r` would become a type of `TGetBlockResult`.